tests/functional/test_broken_stdout.py

```diff
@@ -47,13 +45,16 @@ def test_broken_stdout_pipe(deprecated_python: bool) -> None:
     assert returncode == _BROKEN_STDOUT_RETURN_CODE
 
 
-def test_broken_stdout_pipe__log_option(deprecated_python: bool, tmpdir: Path) -> None:
+def test_broken_stdout_pipe__log_option(
+    deprecated_python: bool,
+    tmp_path: pathlib.Path,
+) -> None:
     """
     Test a broken pipe to stdout when --log is passed.
     """
-    log_path = os.path.join(str(tmpdir), "log.txt")
+    log_path = tmp_path.joinpath("log.txt")
     stderr, returncode = setup_broken_stdout_test(
-        ["pip", "--log", log_path, "list"],
+        ["pip", "--log", str(log_path), "list"],
```

Comment:
Maybe use `os.fspath` rather than str? There's no practical difference as we control the argument's type, but it expresses the intent better.

Reply:
Hmm, somehow I misremembered and thought it’s not available on 3.7. If we use `fspath` maybe it’s much easier to call it in `pip()` instead of littering the call everywhere. I’ll look into it.

Reply:
Was added in 3.6 -- https://docs.python.org/3/library/os.html#os.fspath